flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -759,31 +791,49 @@ public boolean contains(ConfigOption<?> configOption) {
         }
 
         synchronized (this.confData) {
+            if (canBePrefixMap) {
+                removePrefixMap(this.confData, key);
+            }
```

Comment:
No this is not symmetric, because we cannot control `map.k1 = 1` as we have no information about the ConfigOption type. Nevertheless, I would like to avoid confusion by having both `map.k1` `map` in the key space at the same time if we have the possibility to clean up. The method is consistent with `removeConfig`.

Reply:
Couldn't we also merge `k1` into `map` and have it symmetric? I'd be also fine to leave as is if properly documented.

Reply:
I wouldn't start with the merging business. This makes things unnecessary complicated and error prone. We don't have many map types in Flink and most of them are expressed via SQL plain string options.